internal/driver/onvifclient.go

```diff
@@ -418,25 +417,12 @@ func (onvifClient *OnvifClient) callSubscribeCameraEventFunction(resourceName, s
 
 // callGetSnapshotFunction returns a snapshot from the camera as a slice of bytes
 // The implementation can refer to https://github.com/edgexfoundry/device-camera-go/blob/5c4f34d1d59b8e25e1a6316661d463e2495d45fe/internal/driver/onvifclient.go#L119
-func (onvifClient *OnvifClient) callGetSnapshotFunction() ([]byte, errors.EdgeX) {
-	// Get the token from the profile
-	respContent, edgexErr := onvifClient.callOnvifFunction(onvif.MediaWebService, onvif.GetProfiles, nil)
-	if edgexErr != nil {
-		return nil, errors.NewCommonEdgeXWrapper(edgexErr)
-	}
-	profilesResp, ok := respContent.(*media.GetProfilesResponse)
-	if !ok {
-		return nil, errors.NewCommonEdgeX(errors.KindServerError, fmt.Sprintf("invalid GetProfilesResponse of type %T for the camera %s", respContent, onvifClient.DeviceName), nil)
-	}
-	if len(profilesResp.Profiles) == 0 {
-		return nil, errors.NewCommonEdgeX(errors.KindServerError, "no onvif profiles found", nil)
-	}
-	requestData, edgexErr := snapshotUriRequestData(profilesResp.Profiles[0].Token)
-	if edgexErr != nil {
-		return nil, errors.NewCommonEdgeXWrapper(edgexErr)
+func (onvifClient *OnvifClient) callGetSnapshotFunction(profileToken []byte) ([]byte, errors.EdgeX) {
+	if len(profileToken) == 0 {
+		return nil, errors.NewCommonEdgeX(errors.KindContractInvalid, "no profile token in request", nil)
```

Comment:
what happens if you leave this code out? Don't most onvif commands let the camera respond an error if the user input is bad?

Reply:
In my case, I tricked the code by sending `e30=` which is base64 for `{}` (empty json). It responds with a valid image. And I am OK with that, if that is how the camera treats the command.

Reply:
By this I mean you should remove the check for data, as it is possible that field is optional (either for some cameras or all cameras)